net/HttpRequest.hpp

```diff
@@ -678,6 +678,13 @@ class Response final
         saveBodyToMemory();
     }
 
+    Response(const Response& response)
+        : _statusLine(std::move(response._statusLine)),
+          _header(std::move(response._header)),
+          _body(std::move(response._body))
```

Comment:
Unfortunately, this won't work.

This is a copy constructor, but, instead of copying, it is moving the members of its argument. A copy constructor should only copy and should not leave the argument in "unspecified state" (as the standard calls moved-from objects). If we want a move constructor, then we should add it (we would need `Response&& response` parameter).